src/apps/integrations/crud/integrations.py

```diff
@@ -21,14 +21,22 @@ async def create(self, schema: IntegrationsSchema) -> IntegrationsSchema:
             new_integrations = await self._create(schema)
         except IntegrityError:
             raise IntegrationsConfigurationsTypeAlreadyAssignedToAppletError(
-                type=schema.type, applet_id=schema.applet_id
+                integration_type=schema.type, applet_id=schema.applet_id
             )
         return new_integrations
 
-    async def retrieve_by_applet_and_type(self, applet_id: uuid.UUID, type: str) -> IntegrationsSchema:
+    async def retrieve_by_applet_and_type(self, applet_id: uuid.UUID, integration_type: str) -> IntegrationsSchema:
         query = select(IntegrationsSchema)
         query = query.where(IntegrationsSchema.applet_id == applet_id)
-        query = query.where(IntegrationsSchema.type == type)
+        query = query.where(IntegrationsSchema.type == integration_type)
         query = query.limit(1)
         result: Result = await self._execute(query)
         return result.scalars().first()
+
+    async def delete_by_id(self, id_: uuid.UUID):
+        """Delete integrations by id."""
+
+        query = update(IntegrationsSchema)
+        query = query.where(IntegrationsSchema.id == id_)
+        query = query.values(is_deleted=True)
+        await self._execute(query)
```

Comment:
`await self._delete(id=id_)`

Also makes sense to rename `id_`

Reply:
In this implementation we do not have the integrations_id but just the unique key combination applet_id and type which is why those two are used to delete the corresponding integration.
I could update the retrieval to get the integrations id and delete the way you are suggesting here.

Reply:
This is similar to this comment https://github.com/ChildMindInstitute/mindlogger-backend-refactor/pull/1583#discussion_r1769668691
If agreed I can return the integration_id in the retrieval endpoints and so we will be able to delete by its id.